lib/sneakers/handlers/maxretry.rb

```diff
@@ -165,9 +165,16 @@ def failure_count(headers)
         if headers.nil? || headers['x-death'].nil?
           0
         else
-          headers['x-death'].select do |x_death|
+          x_death_array = headers['x-death'].select do |x_death|
             x_death['queue'] == @worker_queue_name
-          end.count
+          end
+          if x_death_array.count > 0 && x_death_array.first['count']
+            # Newer versions of RabbitMQ return headers with a count key
+            x_death_array.inject(0) {|sum, x_death| sum + x_death['count']}
+          else
+            # Older versions return a separate x-death header for each failure
+            x_death_array.count
+          end
```

Comment:
@justin-yesware I updated the logic here so it's more clear when it's using the new vs legacy code.  I also updated it to do a sum of the counts in case there were failures for multiple reasons.